src/Grpc.AspNetCore.Server/Internal/CallHandlers/ClientStreamingServerCallHandler.cs

```diff
@@ -56,11 +69,46 @@ protected override async Task HandleCallAsyncCore(HttpContext httpContext)
                 serverCallContext.Initialize();
 
                 service = activator.Create();
+                
+                if (_interceptors == null)
+                {
+                    response = await _invoker(
+                        service,
+                        new HttpContextStreamReader<TRequest>(serverCallContext, Method.RequestMarshaller.Deserializer),
+                        serverCallContext);
+                }
+                else
+                {
+                    ClientStreamingServerMethod<TRequest, TResponse> resolvedInvoker = delegate (IAsyncStreamReader<TRequest> resolvedRequestStream, ServerCallContext resolvedContext)
+                    {
+                        return _invoker(
+                        service,
+                        resolvedRequestStream,
+                        resolvedContext);
+                    };
+
+                    for (var i = _interceptors.Count - 1; i >= 0; i--)
+                    {
+                        // TODO: Need to check if there's a way to avoid reflection
+                        var registration = _interceptors[i];
+                        var activatorType = typeof(IGrpcInterceptorActivator<>).MakeGenericType(registration.Type);
+                        var interceptorActivator = httpContext.RequestServices.GetRequiredService(activatorType);
+                        var createMethod = activatorType.GetMethod("Create");
```

Comment:
Add a non-generic `IGrpcInterceptorActivator` interface that `IGrpcInterceptorActivator<T>` inherits from. It would have non-generic Create and Release methods. You can cast to the non-generic version and use the methods without reflection.